app/controllers/investigations/products_controller.rb

```diff
@@ -40,13 +40,17 @@ def create
   end
 
   def remove
-    authorize @investigation, :update?
+    authorize @investigation, :remove_product?
+
+    @supporting_information = @product.supporting_information.select { |si| si.investigation == @investigation }
```

Comment:
This should be quicker.
```suggestion
    @supporting_information = @product.supporting_information.where(investigation_id: @investigation.id)
```

Reply:
Unfortunately `supporting_information` returns an `Array` of differing objects rather than a relation: https://github.com/OfficeForProductSafetyAndStandards/product-safety-database/blob/master/app/models/product.rb#L73-L75 (_undefined method `where' for []:Array (NoMethodError)_) 😞